functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
```

Comment:
Added another check as it caught me out while testing. Now checking that the backup is from a SQL  versison we can restore onto the target (same version or 2  major revisions behind).

So here's an update diagram. Might be easier to shove this into github for a while
[Restore-flow1.pdf](https://github.com/sqlcollaborative/dbatools/files/752839/Restore-flow1.pdf)